src/com/tony/server/Main.java

```diff
@@ -38,16 +36,22 @@ public static void main(String[] args) {
         return uriList;
     }
 
-    public static Router createCobSpecRouter() {
+    public static Router createCobSpecRouter(String directoryPath) {
         Router router = new Router();
-        router.addRoute(new Request("GET", "/"), new GetResponse());
-        router.addRoute(new Request("POST", "/form"), new PutPostResponse());
-        router.addRoute(new Request("PUT", "/form"), new PutPostResponse());
-        router.addRoute(new Request("GET", "/method_options"), new GetResponse());
-        router.addRoute(new Request("HEAD", "/method_options"), new GetResponse());
-        router.addRoute(new Request("POST", "/method_options"), new PutPostResponse());
-        router.addRoute(new Request("PUT", "/method_options"), new PutPostResponse());
+        router.addRoute(new Request("GET", "/"), new FileContentResponse(directoryPath + "/"));
+        router.addRoute(new Request("GET", "/form"), new FileContentResponse(directoryPath + "/form"));
+        router.addRoute(new Request("POST", "/form"), new PutPostResponse(directoryPath + "/form"));
+        router.addRoute(new Request("PUT", "/form"), new PutPostResponse(directoryPath + "/form"));
+        router.addRoute(new Request("DELETE", "/form"), new DeleteResponse(directoryPath + "/form"));
+        router.addRoute(new Request("GET", "/method_options"), new FileContentResponse(directoryPath + "/method_options"));
+        router.addRoute(new Request("HEAD", "/method_options"), new HeadResponse());
+        router.addRoute(new Request("POST", "/method_options"), new PutPostResponse(directoryPath + "/method_options"));
+        router.addRoute(new Request("PUT", "/method_options"), new PutPostResponse(directoryPath + "/method_options"));
         router.addRoute(new Request("OPTIONS", "/method_options"), new OptionsResponse(router, "/method_options"));
```

Comment:
No need to pass the entire router to OptionsResponse here, it only needs the allowed methods so pass those in instead